src/flexmock/api.py

```diff
@@ -311,75 +313,62 @@ def updated(self: Any) -> Any:
     def _create_mock_method(self, name: str) -> Callable[..., Any]:
         def _handle_exception_matching(expectation: Expectation) -> None:
             # pylint: disable=misplaced-bare-raise
-            return_values = _getattr(expectation, "_return_values")
-            if return_values:
-                raised, instance = sys.exc_info()[:2]
-                assert raised, "no exception was raised"
-                message = "%s" % instance
-                expected = return_values[0].raises
-                if not expected:
-                    raise
-                args = return_values[0].value
-                expected_instance = expected(*args["kargs"], **args["kwargs"])
-                expected_message = "%s" % expected_instance
-                if inspect.isclass(expected):
-                    if expected is not raised and expected not in raised.__bases__:
-                        raise ExceptionClassError("expected %s, raised %s" % (expected, raised))
-                    if args["kargs"] and isinstance(args["kargs"][0], RE_TYPE):
-                        if not args["kargs"][0].search(message):
-                            raise (
-                                ExceptionMessageError(
-                                    'expected /%s/, raised "%s"'
-                                    % (args["kargs"][0].pattern, message)
-                                )
-                            )
-                    elif expected_message and expected_message != message:
+            return_values = expectation._return_values
+            if not return_values:
+                raise
+            raised, instance = sys.exc_info()[:2]
+            assert raised, "no exception was raised"
+            message = "%s" % instance
+            expected = return_values[0].raises
+            if not expected:
+                raise
+            args = return_values[0].value
+            expected_message = "%s" % expected(*args["kargs"], **args["kwargs"])
+            if inspect.isclass(expected):
+                if expected is not raised and expected not in raised.__bases__:
+                    raise ExceptionClassError("expected %s, raised %s" % (expected, raised))
```

Comment:
Should we do all the f-string in a separate PR ?

Reply:
Yeah that might be better. I was thinking about opening a PR to update tests to check also exception messages. Tests missed one regression in my refactoring PR because we just check the exception type. Same time I can update exception messages and use f-strings.

Adarsh, if you agree, just resolve all my f-string suggestions.

Reply:
Yeah I think we can do the f-string changes in a different unit of work.